datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.datalake.service.resize.recovery;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.exception.BadRequestException;
+import com.sequenceiq.sdx.api.model.SdxRecoverableResponse;
+import com.sequenceiq.sdx.api.model.SdxRecoveryResponse;
+
+@Service
+/**
+ * Provides entrypoint for recovery of failed SDX resize.
+ *
+ * Two public methods:
+ * <ul>
+ *     <li>{@code validateRecovery} checks if resize recovery is appropriate</li>
+ *     <li>{@code triggerRecovery} checks if resize recovery is appropriate, and attempts the recovery</li>
+ * </ul>
+ */
+public class ResizeRecoveryService {
+
+    /**
+     * Determines if it's possible to run a <em>Resize</em> recovery.
+     * @return true if resize recovery is possible.
+     */
+    public boolean canRecover() {
```

Comment:
nit: I like the Javadocs you're adding. But this method is missing from the list of public methods.

Reply:
Thanks, I think the list of public methods is actually bad, they should be annotating each method individually.

I've tuned up the javadocs.